lib/events.js

```diff
@@ -425,9 +425,9 @@ EventEmitter.prototype.listeners = function listeners(type) {
     if (!evlistener)
       ret = [];
     else if (typeof evlistener === 'function')
-      ret = [evlistener];
+      ret = [evlistener.listener || evlistener];
     else
-      ret = arrayClone(evlistener, evlistener.length);
```

Comment:
I would expect declaring `unwrapListeners` within `listeners` (I believe that's what you're suggesting?) to result in a performance degradation.